Feature: Ability to add custom code to bootup

grahambrown11 opened the pull request:
## Description
The ability to add customisations in 2 places: in the beginning after the config is loaded and once the user is loaded.

## Motivation and Context
1. We need to execute some code after the config is loaded in order to get better reporting from our application performance monitoring system, so following the `custom/include` pattern: if
 `custom/include/entryPoint.php` exists it'll be `required` .
2. Implements a `after_load_user` hook so custom code can be executed on the loaded user before continuing with the application logic, this is different from the `after_login` as that only fires once when the user performs the login, where this will fire for every request when the user is retrieved from the session

## How To Test This
Both need to be tested manually as there are no unit tests available for the `entryPoint.php` and `SugarApplicationTest::testloadUser` is commented out due to `die` logic...
1. Add custom code in `custom/include/entryPoint.php` and make sure it gets executed.
2. Adding an application hook registration for `after_load_user` and ensuring the function is executed
```php
if (!isset($hook_array['after_load_user'])) {
    $hook_array['after_load_user'] = [];
}
$hook_array['after_load_user'][] = [
    1,
    'Check User',
    'custom/<HookClass>.php',
    '<HookClass>',
    '<checkUserFunction>'
];
```
I've manually tested these both with the file and hook present and not present.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
<!--- Go over all the following points and check all the boxes that apply. --->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! --->
- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [x] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

isleshocky77 committed:
Adding the abily to relate a Note with a Campaign

isleshocky77 committed:
PSR-2 Convert tabs to spaces

isleshocky77 committed:
Code Format after tab -> spaces conversion

isleshocky77 committed:
Remove duplicate array key (clean warning)

Dillon-Brown committed:
Remove fullcalendar

Dillon-Brown committed:
Fixed #3057 - FullCalendar update to version 3.1

Dillon-Brown committed:
Fixed #4511 - Upgrade of TinyMCE to latest 4.7.x version

Dillon-Brown committed:
Remove tinyMCE

Dillon-Brown committed:
Fixed #3100 - Upgrade for HTML Purifier [Request]

Dillon-Brown committed:
Merge pull request #6714 from gymad/develop_elasticpager_fix

fix last page when modulus is zero

Dillon-Brown committed:
Merge branch 'develop' into tiny_mce

Dillon-Brown committed:
Merge pull request #6730 from Dillon-Brown/issue-6720

Fixed #6720 - Remove old recaptcha files

Dillon-Brown committed:
Merge branch 'develop' into fullcalendar

Dillon-Brown committed:
Merge branch 'develop' into fullcalendar

isleshocky77 committed:
Cleaning up the clean cache command

* Update command to allow for cache directory specified in config
* Remove warnings/errors when no cached directory present
* Update to have namespace "cache" so we can add other cache commands later
  * Leave alias for current command
* Use provided toolsets required by robo for implementation
* Print cleaned directories
* Confirm before cleaning directory
  * Allow for force override

Dillon-Brown committed:
Merge branch 'develop' into fix-clean-cache-command-config-cache-dir

isleshocky77 committed:
Code formatting

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into feature/campaign-notes

# Conflicts:
#	modules/Campaigns/Campaign.php
#	modules/Campaigns/views/view.detail.php
#	modules/Notes/vardefs.php

Dillon-Brown committed:
Reformat

Dillon-Brown committed:
Fix test

samus-aran committed:
Merge pull request #4606 from isleshocky77/feature/campaign-notes

Add ability to have Notes on a Campaign

Dillon-Brown committed:
Formatting

Dillon-Brown committed:
Merge pull request #6802 from isleshocky77/fix-clean-cache-command-config-cache-dir

Allow for clean cache command to use config'd cache_dir (plus clean up)

Dillon-Brown committed:
Remove PHP4 constructors

Dillon-Brown committed:
Nusoap PHP4 style constructors

Dillon-Brown committed:
More PHP4 style constructors

Dillon-Brown committed:
Review comments

cherrador committed:
Update modules/AOR_Charts/lib/pChart/class/pRadar.class.php

Co-Authored-By: Dillon-Brown <26431166+Dillon-Brown@users.noreply.github.com>

lazka committed:
Fixes #7111 - composer: include mpdf/mpdf 6.1 and drop in-tree version (5.7)

Remove the included version of mpdf (in modules/AOS_PDF_Templates/PDF_Lib) and
add it to composer instead.

This also updates the version from 5.7.1 to 6.1.0 to get some fixes that were already
applied manually to the version in SuiteCRM. There are many newer versions available,
but they require more changes and this is a good first step (and this potentially
could be backported to 7.10)

The version in tree had one manual fix applied (d1279cad9c717) which is included in
6.1.0 already, so the upstream version can be used as is.

The required changes to make things compatible with with 6.1 were
* mPDF_exception is now called MpdfException
* SetAutoFont() no longer exists, the docs suggest to use autoLangToFont instead
  to get the same results as before.

Fixes #7111

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into fullcalendar

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/hotfix' into develop_merge

code-ph0y committed:
Merge pull request #6702 from Dillon-Brown/fullcalendar

Fixed #3057 - FullCalendar update to version 3.1

Dillon-Brown committed:
Merge branch 'develop' into develop_merge

samus-aran committed:
Merge pull request #6703 from Dillon-Brown/htmlpurifier

Fixed #3100 - Upgrade for HTML Purifier [Request]

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/master' into develop_merge

Dillon-Brown committed:
Merge remote-tracking branch 'origin/develop_merge' into develop_merge

# Conflicts:
#	composer.json

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Merge pull request #7131 from Dillon-Brown/develop_merge

Develop merge

Dillon-Brown committed:
Merge branch 'develop' into remove_constructors

Dillon-Brown committed:
Add pclzip/pclzip to composer

Dillon-Brown committed:
Remove pclzip references

Dillon-Brown committed:
Remove pclZip from include

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Remove zip_utils.php

Dillon-Brown committed:
Reformat php_zip_utils.php

Dillon-Brown committed:
Update composer

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into tiny_mce

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Update composer

Dillon-Brown committed:
Update tinymce references

Dillon-Brown committed:
Add tedivm/jshrink to composer

Dillon-Brown committed:
Remove Minifier.php

Dillon-Brown committed:
Fixed #7027 - Use a maintained library to minify JS

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into fix-mpdf-composer

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Update composer

Dillon-Brown committed:
PHP 5.6 compat

code-ph0y committed:
Merge pull request #6699 from Dillon-Brown/tiny_mce

Fixed #4511 - Upgrade of TinyMCE to latest 4.9.x version

Dillon-Brown committed:
Merge branch 'develop' into fix-mpdf-composer

Dillon-Brown committed:
Merge pull request #7118 from lazka/fix-mpdf-composer

Fixes #7111 - composer: include mpdf/mpdf 6.1 and drop in-tree version (5.7)

code-ph0y committed:
Merge branch 'develop' into issue-7027

cameronblaikie committed:
Merge pull request #7295 from Dillon-Brown/issue-7027

Fixed #7027 - Use a maintained library to minify JS

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into feature/plcZip

# Conflicts:
#	composer.json
#	composer.lock

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into feature/plcZip

# Conflicts:
#	composer.lock

Dillon-Brown committed:
Update composer.lock

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into remove_constructors

# Conflicts:
#	modules/AOS_PDF_Templates/PDF_Lib/classes/bmp.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/cssmgr.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/directw.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/form.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/gif.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/grad.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/indic.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/svg.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/tocontents.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/ttfontsuni.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/wmf.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdf.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/fpdi_pdf_parser.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/pdf_context.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/pdf_parser.php

grahambrown11 committed:
Ability to add custom code to bootup

entryPoint to load entryPoint in custom folder
Implement a `after_load_user` hook so custom code can be executed on the loaded user before continuing with the application logic

Dillon-Brown committed:
Merge pull request #7287 from Dillon-Brown/feature/plcZip

Remove PclZip

samus-aran commented:
Hey @grahambrown11 Thanks for your contribution. We'll have a wee look :)

connorshea committed:
Remove HTMLPurifier ConfigForm.js file.

This probably isn't necessary and can be removed?

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/hotfix' into develop_merge_1

# Conflicts:
#	composer.lock
#	files.md5
#	include/HTMLPurifier/HTMLPurifier.standalone.php
#	include/HTMLPurifier/standalone/HTMLPurifier/Lexer/PH5P.php
#	include/SugarCharts/Jit/FlashCanvas/canvas2png.js
#	include/SugarCharts/Jit/FlashCanvas/flashcanvas.js
#	include/SugarCharts/Jit/js/Jit/jit.js
#	include/SugarCharts/Jit/js/sugarCharts.js
#	include/pclzip/pclzip.lib.php
#	include/utils/zip_utils.php
#	lib/Robo/Plugin/Commands/CleanCacheCommands.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/barcode.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/bmp.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/cssmgr.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/directw.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/form.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/gif.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/grad.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/indic.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/meter.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/svg.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/tocontents.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/ttfontsuni.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/ttfontsuni_analysis.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/wmf.php
#	modules/AOS_PDF_Templates/PDF_Lib/compress.php
#	modules/AOS_PDF_Templates/PDF_Lib/config.php
#	modules/AOS_PDF_Templates/PDF_Lib/config_cp.php
#	modules/AOS_PDF_Templates/PDF_Lib/config_fonts.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ccourier.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ccourierb.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ccourierbi.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ccourieri.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/chelvetica.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/chelveticab.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/chelveticabi.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/chelveticai.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/csymbol.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ctimes.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ctimesb.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ctimesbi.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/ctimesi.php
#	modules/AOS_PDF_Templates/PDF_Lib/font/czapfdingbats.php
#	modules/AOS_PDF_Templates/PDF_Lib/graph.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/CJKdata.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/functions.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_bn_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_gu_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_hi_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_kn_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_ml_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_or_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_pa_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_ta_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/ind_te_1_001.volt.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/out.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/subs_core.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/subs_win-1252.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/upperCase.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdf.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/filters/FilterASCII85.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/filters/FilterLZW.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/fpdi_pdf_parser.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/pdf_context.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdfi/pdf_parser.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/de.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/en.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/es.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/fi.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/fr.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/it.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/nl.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/pl.php
#	modules/AOS_PDF_Templates/PDF_Lib/patterns/sv.php
#	modules/AOS_PDF_Templates/PDF_Lib/utils/UnicodeRanges.php
#	modules/AOS_PDF_Templates/PDF_Lib/utils/font_collections.php
#	modules/AOS_PDF_Templates/PDF_Lib/utils/font_coverage.php
#	modules/AOS_PDF_Templates/PDF_Lib/utils/font_dump.php
#	modules/AOS_PDF_Templates/PDF_Lib/utils/font_names.php
#	modules/Emails/include/ListView/DeleteEmailAction.js
#	modules/UpgradeWizard/silentUpgrade_step1.php

Dillon-Brown committed:
Update composer.lock

samus-aran committed:
Merge pull request #7674 from Dillon-Brown/develop_merge_1

Develop merge

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into remove_constructors

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/develop' into remove_constructors

# Conflicts:
#	include/nusoap/class.soap_parser.php
#	include/nusoap/class.xmlschema.php

grahambrown11 commented:
Any reason this has not been merged?
I understand if it's priorities, just want to make sure I've done nothing wrong...

samus-aran commented:
> Any reason this has not been merged?
> I understand if it's priorities, just want to make sure I've done nothing wrong...

From what I can see you haven't done anything wrong. Just circulating if there is any issues. :)

grahambrown11 commented:
> > Any reason this has not been merged?
> > I understand if it's priorities, just want to make sure I've done nothing wrong...
> 
> From what I can see you haven't done anything wrong. Just circulating if there is any issues. :)

Thanks for the feedback

HamburgischerVereinSeefahrt committed:
The current SuiteP Theme does currently not allow to differentiate between different modules in the html code. While this is not necessary for the standard SuiteP layout, it would allow for tiny customization with css here and there. The present commit adds a "data-module" tag to the SuiteP pagecontent div and "data-field" tags in the edit and detail views. This allows developers to easily customize the appearance of individual views by simply adding custom css code, e.g.:
div#pagecontent[data-module='Accounts'] .edit-view-row-item[data-field='assigned_user_name'] {color: red;}

samus-aran committed:
Merge pull request #6750 from Dillon-Brown/remove_constructors

Remove PHP4 style constructors

Dillon-Brown committed:
Merge pull request #7606 from connorshea/remove-orphaned-html-purifier-file

Remove HTMLPurifier ConfigForm.js file.

connorshea committed:
Remove dead code.

As far as I've been able to tell, this code is entirely unused. So I've removed it.

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/hotfix' into develop_merge_4

# Conflicts:
#	codeception.yml
#	composer.json
#	composer.lock
#	include/HTMLPurifier/HTMLPurifier.standalone.php
#	include/HTMLPurifier/standalone/HTMLPurifier/Printer/ConfigForm.php
#	include/MVC/View/SugarView.php
#	include/nusoap/class.nusoap_base.php
#	include/nusoap/class.soap_transport_http.php
#	include/nusoap/class.soapclient.php
#	include/nusoap/class.wsdl.php
#	include/nusoap/class.wsdlcache.php
#	include/parsecsv.lib.php
#	modules/AOR_Charts/lib/pChart/class/pBarcode128.class.php
#	modules/AOR_Charts/lib/pChart/class/pBarcode39.class.php
#	modules/AOR_Charts/lib/pChart/class/pCache.class.php
#	modules/AOR_Charts/lib/pChart/class/pData.class.php
#	modules/AOR_Charts/lib/pChart/class/pImage.class.php
#	modules/AOR_Charts/lib/pChart/class/pSpring.class.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/barcode.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/cssmgr.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/gif.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/grad.php
#	modules/AOS_PDF_Templates/PDF_Lib/classes/ttfontsuni.php
#	modules/AOS_PDF_Templates/PDF_Lib/compress.php
#	modules/AOS_PDF_Templates/PDF_Lib/graph.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/functions.php
#	modules/AOS_PDF_Templates/PDF_Lib/includes/no_image.jpg
#	modules/AOS_PDF_Templates/PDF_Lib/mpdf.php
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_diagonals-thick_90_eeeeee_40x40.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_glass_100_e4f1fb_1x400.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_glass_50_3baae3_1x400.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_glass_80_d7ebf9_1x400.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_highlight-hard_100_f2f5f7_1x100.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-bg_highlight-hard_70_000000_1x100.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-icons_2694e8_256x240.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-icons_2e83ff_256x240.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-icons_3d80b3_256x240.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-icons_72a7cf_256x240.png
#	modules/Calendar/fullcalendar/lib/cupertino/images/ui-icons_ffffff_256x240.png
#	modules/Employees/Employee.php

Dillon-Brown committed:
Update composer.lock

connorshea committed:
Remove deprecated DBHelper class.

This hasn't been used for over 6 years. Time to clean it up.

connorshea committed:
Remove FreeTDSHelper.

Deprecated for years.

connorshea committed:
Remove MssqlHelper.

Deprecated for many years.

The method referenced in SqlsrvManager.php's comments doesn't seem to
exist anywhere, so I've removed the reference entirely.

connorshea committed:
Remove MysqlHelper.

It's been deprecated for many years now.

cameronblaikie committed:
Merge pull request #7844 from HVStechnik/feature/SuiteP-add-identifying-html-module-and-view-tag

SuiteP: Add html data tags to allow module and field identification

Dillon-Brown committed:
Merge pull request #8074 from connorshea/remove-deprecated-helper-files

Remove deprecated helper files

Dillon-Brown committed:
Merge pull request #8084 from connorshea/activity-subpanel-view-smarty

Remove dead code: Re:Remastered

connorshea committed:
Remove HTMLSax3.

This was only ever used as part of HTMLPurifier (as an experimental
parser), as far as I can tell. Since HTMLPurifier is no longer vendored
in the repository, this library no longer needs to be vendored either.

Dillon-Brown committed:
Merge pull request #8088 from connorshea/remove-pear-html-sax

Remove XML_HTMLSax3

Dillon-Brown committed:
Merge pull request #8066 from Dillon-Brown/develop_merge_4

Develop merge

connorshea committed:
Remove all deprecated Smarty files.

Smarty is included via Composer now, so these files can be removed. If
anyone is requiring these files in their custom code, they
should make sure to remove those requires.

All Smarty plugins remain as-is.

connorshea committed:
Remove SugarXHProf.

The XHProf project has been abandoned for 4 years, and
the version vendored here is much older than even that. It's
undocumented, unmaintained, and it doesn't seem like it even works
on PHP 7.x.

"The best way to choose what to keep and what to throw away
is to take each item in one’s hand and ask: 'Does this spark joy?'
If it does, keep it. If not, dispose of it."

- Marie Kondō

connorshea committed:
Remove deprecated encrypt_password function for User.

It's not used anywhere else in the codebase besides the test
suite, should be safe to remove.

connorshea committed:
Remove deprecated JSON functions.

These have been deprecated for over six years now.

There was only one use of encodeReal, so I've replaced it.

connorshea committed:
Remove unused, deprecated functions from DBManager.php.

This leaves getHelper() for now because it's still used in a
few places. As far as I can tell, all of these have been deprecated
since at  least the fork from SugarCRM.
None of the removed functions are being used elsewhere in the codebase.

connorshea committed:
Remove deprecated getReminderTime function in CallHelper.

connorshea committed:
Remove remove_redundant_http() from Account.

This code doesn't do anything and hasn't been used for a long time.

Also removes the test for it.

connorshea committed:
Remove get_day_by_index_this_year from SugarDateTime.

This has been deprecated for a long time and is unused.

Dillon-Brown committed:
Merge pull request #8128 from connorshea/remove-deprecated-sugar-date-time-function

Remove get_day_by_index_this_year from SugarDateTime.

Mac-Rae committed:
Merge pull request #8127 from connorshea/remove-remove-redundant-http-function

Remove remove_redundant_http() from Account.

Mac-Rae committed:
Merge pull request #8126 from connorshea/remove-deprecated-call-helper-function

Remove deprecated getReminderTime function in CallHelper.

Mac-Rae committed:
Merge pull request #8125 from connorshea/remove-deprecated-code-from-dbmanager

Remove unused, deprecated functions from DBManager.php.

samus-aran commented:
Hey @grahambrown11 
Sorry for the delayed response. The biggest issue we have is that this hook appears to be from Sugar propriety and we would rather not encourage that behaviour. We can see that based on the recent codebase that its a sound hook and does offer value to the users. So our suggestion is to change the hook's name itself to differentiate it from the current implementation that Sugar has.
We felt that after_user_load would be acceptable and also feel its more inline with the other hook's name structure (state_subject_action)

That be suitable for you and your implementation?

Mac-Rae committed:
Merge pull request #8124 from connorshea/remove-deprecated-json-functions

Remove deprecated JSON functions encodeReal() and decodeReal()

Mac-Rae committed:
Merge pull request #8122 from connorshea/remove-deprecated-encrypt-function

Remove deprecated encrypt_password function for User.

Mac-Rae committed:
Merge pull request #8095 from connorshea/remove-xhprof

Remove SugarXHProf

Mac-Rae committed:
Merge pull request #8093 from connorshea/remove-deprecated-smarty

Remove all deprecated Smarty files.

grahambrown11 committed:
change name according to feedback received

grahambrown11 commented:
> Hey @grahambrown11
> Sorry for the delayed response. The biggest issue we have is that this hook appears to be from Sugar propriety and we would rather not encourage that behaviour. We can see that based on the recent codebase that its a sound hook and does offer value to the users. So our suggestion is to change the hook's name itself to differentiate it from the current implementation that Sugar has.
> We felt that after_user_load would be acceptable and also feel its more inline with the other hook's name structure (state_subject_action)
> 
> That be suitable for you and your implementation?

Sure @samus-aran, sorry for what seemed to be a copy of Sugar propriety function - it kinda was, I was searching the Google for how to implement and came across the Sugar documentation, and saw it was not implemented in Suite, so yes the name was copied but not the code as I have no access to that code. I've updated my branch, and it seems to have updated this PR - hope you find this is order...

connorshea committed:
Remove the deprecated load_menu() function in utils.php

It's been deprecated since SuiteCRM's initial commit and a grep
for load_menu doesn't return anything, so it doesn't seem to be used
anywhere.

connorshea committed:
Remove deprecated functions from utils.php.

Remove get_admin_modules_for_user, get_theme_display, get_themes,
and isValidId.

These are all deprecated and unused.

Dillon-Brown committed:
Merge pull request #8218 from connorshea/remove-deprecated-functions

Remove deprecated functions from utils.php.

Dillon-Brown committed:
Merge pull request #8217 from connorshea/remove-deprecated-load-menu-function

Remove the deprecated load_menu() function in utils.php

grahambrown11 committed:
Merge branch 'develop' of github.com:salesagility/SuiteCRM into feature-bootup-customisation

grahambrown11 committed:
Merge branch 'feature-bootup-customisation' of github.com:grahambrown11/SuiteCRM into feature-bootup-customisation

mattlorimer committed:
Fix #8541 - MySQL Database breaking on special characters

mattlorimer committed:
Roll back backward incompatible change

remove unrequired deprecation log

Dillon-Brown committed:
Update travis config with strict_id_validation

Signed-off-by: Dillon-Brown <dillon.brown@salesagility.com>

samus-aran commented:
Restarting Job :+1:

mattlorimer committed:
Fix strict_id_validation

mattlorimer committed:
7.11.12

Dillon-Brown committed:
Merge remote-tracking branch 'upstream/hotfix' into develop

# Conflicts:
#	composer.json
#	composer.lock
#	files.md5
#	include/HTMLPurifier/HTMLPurifier.standalone.php
#	include/HTMLPurifier/standalone/HTMLPurifier/ConfigSchema/schema/Attr.IDPrefixLocal.txt
#	include/HTMLPurifier/standalone/HTMLPurifier/Lexer/PEARSax3.php
#	include/Pear/XML_HTMLSax3/HTMLSax3.php
#	include/database/DBManager.php
#	include/generic/SugarWidgets/SugarWidgetField.php
#	include/pclzip/pclzip.lib.php
#	include/templates/Template.php
#	install/installCommon.js
#	modules/ACLActions/ACLAction.php
#	modules/AOR_Reports/controller.php
#	modules/AOS_PDF_Templates/PDF_Lib/mpdf.php
#	modules/Accounts/Account.php
#	modules/Activities/SubPanelView.php
#	modules/Calendar/fullcalendar/fullcalendar.css
#	modules/Notes/vardefs.php
#	tests/unit/phpunit/include/utils/zipUtilsTest.php
#	tests/unit/phpunit/modules/Users/UserTest.php

Dillon-Brown committed:
Update composer.lock

Signed-off-by: Dillon-Brown <dillon.brown@salesagility.com>

Dillon-Brown committed:
Fix syntax error with notes vardef ''optimistic_locking''

Signed-off-by: Dillon-Brown <dillon.brown@salesagility.com>

Dillon-Brown committed:
Fix ACLRoleTest.php

Signed-off-by: Dillon-Brown <dillon.brown@salesagility.com>

Dillon-Brown committed:
Revert fullcalendar/fullcalendar removal

Signed-off-by: Dillon-Brown <dillon.brown@salesagility.com>

mattlorimer committed:
Merge pull request #8555 from Dillon-Brown/develop

Develop merge

Dillon-Brown committed:
Revert composerify mPDF change

This is a temporary revert to move mPDF back into the core until the full solution of a backwards compatible interface can be achieved.

Signed-off-by: Dillon-Brown <dillon.brown@salesagility.com>

Dillon-Brown committed:
Merge pull request #8592 from Dillon-Brown/fix/mpdf

Temporarily revert composerify mPDF change

grahambrown11 committed:
Merge branch 'develop' into feature-bootup-customisation

grahambrown11 committed:
Merge branch 'develop' of github.com:salesagility/SuiteCRM into develop

grahambrown11 committed:
Merge branch 'develop' into feature-bootup-customisation

grahambrown11 commented:
I don't understand why the build fails... I made this change so long ago and I believe it was approved just never merged...
Will this ever get merged or should I give up on this PR?
